ddapm_test_agent/trace_snapshot.py

```diff
@@ -459,40 +460,57 @@ def _ordered_span(s: Span) -> OrderedDictType[str, TopLevelSpanValue]:
     s = "[\n"
     while stack:
         prefix, span = stack.pop(0)
 
         # Remove any keys that are not needed for comparison
-        if removed:
-            for key in removed:
-                if key.startswith("meta."):
-                    span["meta"].pop(key[5:], None)
-                elif key.startswith("metrics."):
-                    span["metrics"].pop(key[8:], None)
-                elif key.startswith("span_links.attributes."):
-                    if "span_links" in span:
-                        for link in span["span_links"]:
-                            if "attributes" in link:
-                                link["attributes"].pop(key[22:], None)
-                elif key.startswith("span_links."):
-                    if "span_links" in span:
-                        for link in span["span_links"]:
-                            link.pop(key[11:], None)  # type: ignore
-                elif key.startswith("span_events.attributes."):
-                    if "span_events" in span:
-                        for event in span["span_events"]:
-                            if "attributes" in event:
-                                event["attributes"].pop(key[23:], None)
-                elif key.startswith("span_events."):
-                    if "span_events" in span:
-                        for event in span["span_events"]:
-                            event.pop(key[12:], None)  # type: ignore
-                else:
-                    span.pop(key, None)  # type: ignore
+        for key in removed:
+            if key.startswith("meta."):
+                span["meta"].pop(key[5:], None)
+            elif key.startswith("metrics."):
+                span["metrics"].pop(key[8:], None)
+            elif key.startswith("span_links.attributes."):
+                if "span_links" in span:
+                    for link in span["span_links"]:
+                        if "attributes" in link:
+                            link["attributes"].pop(key[22:], None)
+            elif key.startswith("span_links."):
+                if "span_links" in span:
+                    for link in span["span_links"]:
+                        link.pop(key[11:], None)  # type: ignore
+            elif key.startswith("span_events.attributes."):
+                if "span_events" in span:
+                    for event in span["span_events"]:
+                        if "attributes" in event:
+                            event["attributes"].pop(key[23:], None)
+            elif key.startswith("span_events."):
+                if "span_events" in span:
+                    for event in span["span_events"]:
+                        event.pop(key[12:], None)  # type: ignore
+            else:
+                span.pop(key, None)  # type: ignore
+
+        if attribute_regex_replaces:  # only walk if we actually have something to replace
+            _walk_span_attributes_with_regex_replaces(span, attribute_regex_replaces)  # type: ignore
```

Comment:
what is the ignore here for?

does `_walk_span_attributes_with_regex_replaces` need to take a `Span` instead?

or else, we can cast the span to a `Dict[str, Any]`

```suggestion
            _walk_span_attributes_with_regex_replaces(typing.cast(Dict[str, Any], span), attribute_regex_replaces)
```

I didn't look too deeply, but I assume this is the error?

Reply:
Yes :-)